lms/views/api/canvas/authorize.py

```diff
@@ -127,15 +128,24 @@ def oauth2_redirect(request):
     request_method="GET",
     route_name="canvas_api.oauth.authorize",
     renderer="lms:templates/api/oauth2/redirect_error.html.jinja2",
+    tm_active=True,
```

Comment:
Can you clarify why `tm_active` is needed? My understanding from reading the code is that `queue_event` doesn't write to the DB directly but instead dispatches a Celery task which will write to the DB outside of the current event.